### ISSUE TITLE
Enabling the use of QML for Tiled extensions

### ISSUE THREAD

**Issue by bjorn** [mapeditor/tiled#4126]
As of Tiled 1.11, extensions can be written in plain JavaScript files or as JavaScript modules. However, Tiled actually uses a [`QQmlEngine`](https://doc.qt.io/qt-6/qqmlengine.html) internally, which is able to load [QML files](https://doc.qt.io/qt-6/qmlapplications.html) as well.

See [Next generation Tiled extensions](https://github.com/mapeditor/tiled/wiki/Next-generation-Tiled-extensions) for more information about this idea.

When the necessary libraries and plugins are available, [`Qt.createComponent`](https://doc.qt.io/qt-6/qtqml-javascript-dynamicobjectcreation.html) can already be used to instantiate QML types. A nicer integration should be possible when Tiled recognizes QML files and creates [`QQmlComponent`](https://doc.qt.io/qt-6/qqmlcomponent.html#QQmlComponent-1) instances with those files and creates objects from them.

**Comment by NewtronReal** [comment]
Is anyone assigned or working on this feature. If not I  would like to work on the feature.

**Comment by abhinavs1920** [comment]
Hey @NewtronReal! You can go ahead and start working on it. If anything’s unclear, just drop your questions here. @bjorn and other maintainers will review and guide you with the approach.

**Comment by NewtronReal** [comment]
@bjorn @abhinavs1920 

<img width="2539" height="1558" alt="Image" src="https://github.com/user-attachments/assets/f78c2416-2496-4433-a6b4-eea55312381a" />

is this the feature you wanted?

**Comment by NewtronReal** [comment]
I wasn't sure whether this is the feature or not. If this is what you wanted i'll be creating a pull request right away.

**Comment by bjorn** [comment]
@NewtronReal Indeed, it seems like you've made a good start on this feature, and you can read the [wiki page](https://github.com/mapeditor/tiled/wiki/Next-generation-Tiled-extensions) about the general idea and possible extensions we could allow based on QML. However, we have currently somebody working on this as part of the Winter of Code 5.0 (until 16th of March). You can see their work at #4380. While you're free to start your own branch towards this, I will let them finish their project first.

But as I wrote on the [ideas list for GSoC 2026](https://github.com/mapeditor/tiled/wiki/GSoC-2026#next-generation-tiled-extensions), I think even after that this will still leave enough work for a potential GSoC contributor.
